wagtail/wagtailsearch/backends/elasticsearch.py

```diff
@@ -280,8 +280,11 @@ def _process_lookup(self, field, lookup, value):
         if lookup == 'in':
             if isinstance(value, Query):
                 db_alias = self.queryset._db or DEFAULT_DB_ALIAS
-                value = (value.get_compiler(db_alias)
-                         .execute_sql(result_type=SINGLE))
+                value = next(value.get_compiler(db_alias)
+                                  .execute_sql(result_type=MULTI))
```

Comment:
Here `execute_sql` is returning an iterator that yields lists of result rows. Are we confident that we're only ever going to need to look at the first of those lists, as implied by the use of `next`?

(Digging into https://github.com/django/django/blob/308f64462421b09b21ef0dcd9cc3654cc25bceba/django/db/models/sql/compiler.py#L855, it looks like this iterator setup possibly exists in order to support `chunked_fetch`, which is false by default, in which case we're fine. I'm hoping you or @BertrandBordage have already researched this, so I don't have to :-) )

Reply:
For some reason, when running this new code against the updated tests from 411fb6b457ea5ab8735459146c33557af3187011, `execute_sql` is now giving me a list of lists of rows, rather than an iterator of lists of rows, and failing on `next` with `'list' object is not an iterator`. However, I'm still getting back an iterator as expected when I do the same thing via `manage.py shell` against bakerydemo... investigating.

    >>> from wagtail.wagtailsearch.backends import get_search_backend
    >>> backend = get_search_backend('default')
    >>> backend.search('bread', Page.objects.filter(id__in=Page.objects.all().values_list('id', flat=True)))
    multi result: <generator object cursor_iter at 0xb56f6e0c>
    <SearchResults []>

Reply:
ah, it's sqlite (the default for runtests) versus postgresql, isn't it... https://github.com/django/django/blob/308f64462421b09b21ef0dcd9cc3654cc25bceba/django/db/models/sql/compiler.py#L917-L922

Reply:
Now updated in 25727ea to handle list and iterator responses from `execute_sql` equally (and also look beyond the first chunk of chunked resultsets, if we ever have to deal with those).